homeassistant/components/yeelight/config_flow.py

```diff
@@ -60,6 +60,17 @@ async def async_step_dhcp(self, discovery_info):
         self._discovered_ip = discovery_info[IP_ADDRESS]
         return await self._async_handle_discovery()
 
+    async def async_step_zeroconf(self, discovery_info):
+        """Handle discovery from zeroconf."""
+        self._discovered_ip = discovery_info["host"]
+        await self.async_set_unique_id(
+            "{0:#0{1}x}".format(int(discovery_info["name"][-26:-18]), 18)
```

Comment:
Usually we use an `f-string` instead of `.format()`, but I think in this case it might make it less readable.